src/transformers/models/modernbert/modeling_modernbert.py

```diff
@@ -67,45 +67,22 @@ def forward(
         max_seqlen: Optional[int] = None,
     ):
         # (total_nnz, 3, nheads, headdim)
-        total_nnz, three, nheads, headdim = qkv.shape
-        if qkv.is_contiguous():
-            # Call 1 kernel instead of 2 kernels
-            # We need qkv to be contiguous so that when we reshape to combine (3, nheads)
-            # dimensions, we get the same tensor
-            # qk = rearrange(qkv[:, :2], "b_s t h d -> b_s (t h) d")
-            qk = qkv[:, :2].view(total_nnz, -1, headdim)
-            apply_rotary(
-                qk,
-                cos,
-                sin,
-                seqlen_offsets=0,
-                cu_seqlens=cu_seqlens,
-                max_seqlen=max_seqlen,
-                interleaved=False,
-                inplace=True,
-            )
-        else:
-            q, k = qkv[:, 0, :, :], qkv[:, 1, :, :]
-            apply_rotary(
-                q,
-                cos,
-                sin,
-                seqlen_offsets=0,
-                cu_seqlens=cu_seqlens,
-                max_seqlen=max_seqlen,
-                interleaved=False,
-                inplace=True,
-            )
-            apply_rotary(
-                k,
-                cos,
-                sin,
-                seqlen_offsets=0,
-                cu_seqlens=cu_seqlens,
-                max_seqlen=max_seqlen,
-                interleaved=False,
-                inplace=True,
-            )
+        qkv = qkv.contiguous()
```

Comment:
calling contiguous is not free, no?

Reply:
I don't think it is, no. I'd rather keep the faster solution, but I understand @ArthurZucker's desire for simplicity.

Reply:
without additional tests that the performance does not degrade, I would keep the contiguity check.

Reply:
I am not 100% sure and I don't think a test is that hurtful, but it seems like `.contiguous()` is a [no-op if the tensor is already contiguous](https://discuss.pytorch.org/t/does-contiguous-creates-a-copy-of-the-memory/67656/2).
The answer is from 2020, but the [PyTorch documentation seems to agree](https://pytorch.org/docs/stable/generated/torch.Tensor.contiguous.html) and a [check seems to be performed](https://github.com/pytorch/pytorch/blob/ccf35af1421447c6425d6d290cb001b80d93c99b/torch/_refs/__init__.py#L2992).